include/jsontoolkit/jsonschema/walker.h

```diff
@@ -25,30 +25,37 @@ enum class schema_walker_strategy_t {
 using schema_walker_t = std::function<schema_walker_strategy_t(
     const std::string &, const std::unordered_map<std::string, bool> &)>;
 
+enum class schema_walker_type_t { Deep, Flat };
+
 class SchemaWalker {
 public:
   SchemaWalker(const Value &input, const schema_walker_t &walker,
                const schema_resolver_t &resolver,
-               const std::optional<std::string> &default_metaschema);
+               const std::optional<std::string> &default_metaschema,
+               const schema_walker_type_t type);
   inline auto begin() const -> decltype(auto) {
     return this->subschemas.cbegin();
   }
   inline auto end() const -> decltype(auto) { return this->subschemas.cend(); }
 
 private:
   auto walk(const Value &subschema, const schema_walker_t &walker,
-            const schema_resolver_t &resolver, const std::string &metaschema)
-      -> void;
+            const schema_resolver_t &resolver, const std::string &metaschema,
+            const std::size_t level) -> void;
   auto walk_array(const Value &array, const schema_walker_t &walker,
                   const schema_resolver_t &resolver,
-                  const std::string &metaschema) -> void;
+                  const std::string &metaschema, const std::size_t level)
+      -> void;
   auto walk_object(const Value &object, const schema_walker_t &walker,
                    const schema_resolver_t &resolver,
-                   const std::string &metaschema) -> void;
+                   const std::string &metaschema, const std::size_t level)
+      -> void;
   auto walk_schema(const Value &schema, const schema_walker_t &walker,
                    const schema_resolver_t &resolver,
-                   const std::string &metaschema) -> void;
+                   const std::string &metaschema, const std::size_t level)
+      -> void;
   std::vector<std::reference_wrapper<const Value>> subschemas;
```

Comment:
Note to self: we want `flat_subschema_iterator` to be read-write. The `SchemaWalker` class hardcodes a vector of const JSON values, so we will probably need to turn `SchemaWalker` into a templated class that takes the vector element type as a template argument. `subschema_iterator` would pass `const Value` and `flat_subschema_iterator` would pass `Value`.

Reply:
Done!